core/referral/notional_volumes.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2023 Gobalsky Labs Limited
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE.VEGA file and at https://www.mariadb.com/bsl11.
+//
+// Change Date: 18 months from the later of the date of the first publicly
+// available Distribution of this version of the repository, and 25 June 2022.
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by version 3 or later of the GNU General
+// Public License.
+
+package referral
+
+import (
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/num"
+)
+
+type runningVolumes struct {
+	// maxPartyNotionalVolumeByQuantumPerEpoch limits the volume in quantum units
+	// which is eligible each epoch for referral program mechanisms.
+	maxPartyNotionalVolumeByQuantumPerEpoch *num.Uint
+
+	// runningVolumesBySet tracks the running notional volume by referralSets.
+	runningVolumesBySet map[types.ReferralSetID][]*notionalVolume
+}
+
+type notionalVolume struct {
+	epoch uint64
+	value *num.Uint
+}
+
+func (vs *runningVolumes) Add(epoch uint64, setID types.ReferralSetID, volume *num.Uint) {
```

Comment:
does it ever get cleaned up? seems to grow indefinitely unless I've missed the remove.

Reply:
It grows indefinitely for now, we can and should limit. However we don't have a limit do the window_length. I will start a conversation about that.

Reply:
there should definitely be a limit on the window length. for example for the metrics we set it to 100 (to cover a quarter) or so.